### PR TITLE
Handle surrogate pairs in diff

### PR DESCRIPTION
**Pull request by KentoMoriwaki**
fix https://github.com/yjs/yjs/issues/303

Current diff algorithm may produce invalid combinations of surrogate pair.
To avoid this, we can use `Array.from` to calculate the length and index in safe.

This changes may cause performance degradation, so we have an option to implement this in another function. IMO it is better to be able to handle surrogate pairs correctly in all situations.

### CODE REVIEW

**Comment by KentoMoriwaki** [comment]
@dmonad Cloud you take a look?

File: diff.js [hunk 1]
```diff
@@ -39,18 +39,20 @@ import { equalityStrict } from './function.js'
  * @return {SimpleDiff<string>} The diff description.
  */
 export const simpleDiffString = (a, b) => {
+  const aArr = Array.from(a)
+  const bArr = Array.from(b)
```
Comment (tommoor) [inline diff.js:43]: _If_ these computations below to prove to be less performant then perhaps we could compare the length of the array to the original input string and fallback to the original simpler string modifications when possible.

**Comment by KentoMoriwaki** [comment]
I've implemented another version: more performant and simpler, but there is a trade-off. #46 
You can choose which one you prefer.

**Comment by HondryTravis** [comment]
same question!

**Comment by Young6118** [comment]
This worked for me, please merge into the master branch! please! ! ! 🙏

File: diff.js (hunk 1 above)
Comment (Young6118) [reply diff.js:43]: Good job!

**Comment by dmonad** [comment]
Closed in favor of #46

**Pull request closed without merging**